xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp

```diff
@@ -175,21 +308,25 @@ bool CVaapiTexture::Map(CVaapiRenderPicture *pic)
         return false;
       }
 
-      glGenTextures(1, &m_textureY);
-      glBindTexture(m_interop.textureTarget, m_textureY);
-      m_interop.glEGLImageTargetTexture2DOES(m_interop.textureTarget, m_glSurface.eglImageY);
-
-      glGenTextures(1, &m_textureVU);
-      glBindTexture(m_interop.textureTarget, m_textureVU);
-      m_interop.glEGLImageTargetTexture2DOES(m_interop.textureTarget, m_glSurface.eglImageVU);
-
-      glBindTexture(m_interop.textureTarget, 0);
+      m_textureY = ImportImageToTexture(m_glSurface.eglImageY);
+      m_textureVU = ImportImageToTexture(m_glSurface.eglImageVU);
 
       break;
     }
     default:
       return false;
   }
+  
+  return true;
+}
+
+bool CVaapiTexture::Map(CVaapiRenderPicture *pic)
+{
+  if (m_vaapiPic)
+    return true;
+  
+  if (!MapEsh(pic) && !MapImage(pic))
```

Comment:
I don't like this. keep it simple, we either do this or that. we know at the very biginning with path we'll take.